lib/signaling/v2/peerconnection.js

```diff
@@ -488,6 +493,19 @@ class PeerConnectionV2 extends StateMachine {
       this._log.info('ICE reconnected');
     }
 
+    if (iceConnectionState === 'disconnected') {
+      this._iceConnectionMonitor.start(() => {
```

Comment:
@makarandp0 We need to make sure that when we access `PeerConnectionV2#iceConnectionState`, it is set to 'failed' here, even though the actual state is disconnected. We can do something like this:
```js
this._iceConnectionMonitor.start(() => {
  this._isIceConnectionInactive = true;
  ...
});
```
Then, we can modify the `iceConnectionState` getter for PeerConnectionV2 like this:
```js
get iceConnectionState() {
  return this._peerConnection.iceConnectionState === 'disconnected' && this._isIceConnectionInactive
    ? 'failed'
    : this._peerConnection.iceConnectionState;
}
```

Reply:
ah, I missed this comment. makes sense.